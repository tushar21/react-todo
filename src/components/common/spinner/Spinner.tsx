import * as React from "react"
import { connect } from 'react-redux'
import './spinner.css'

interface SpinnerProps{
    loading: boolean;
}

class SpinnerLoader extends React.Component<SpinnerProps>{

    constructor(props:SpinnerProps){
        super(props);
    }
    
    render(){
        const {loading, children} = this.props;
        return ( 
            <div>{loading ? <div className="d-flex justify-content-center" id="spinner"> <div className="spinner-border" role="status"> <span className="sr-only">Loading...</span>
            </div> </div> : children}</div>            
        )
    }    
}


const mapStateToProps = (state:any)=>{
    return {loading: state.users.loading}
}
export default connect(mapStateToProps, null)(SpinnerLoader)


