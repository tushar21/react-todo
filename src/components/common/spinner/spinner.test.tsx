import * as React from 'react';
import {create} from 'react-test-renderer';
import {SpinnerLoader} from './Spinner';

test('default loader should be disabled', () => {
    const component = create(<SpinnerLoader loading={true}>Hello</SpinnerLoader>)
    
    let tree = component.toJSON()
    const loadingSpan  = tree.children[0].children[0].children[0];
    expect(loadingSpan.type).toEqual('span')
    expect(loadingSpan.props.className).toEqual('sr-only')
    expect(loadingSpan.children[0]).toEqual('Loading...')
    expect(tree).toMatchSnapshot();    
})


test('loader should be disabled and content will be displayed', () => {
    const content = 'Hello';
    const component = create(<SpinnerLoader loading={false}>{content}</SpinnerLoader>)    
    let tree = component.toJSON()
    console.log(tree, 'tree')
    expect(tree.type).toEqual('div')
    expect(tree.children[0]).toEqual(content)
    expect(tree).toMatchSnapshot();    
})