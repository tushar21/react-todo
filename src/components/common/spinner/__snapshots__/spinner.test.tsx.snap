// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default loader should be disabled 1`] = `
<div>
  <div
    className="d-flex justify-content-center"
    id="spinner"
  >
    <div
      className="spinner"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`loader should be disabled and content will be displayed 1`] = `
<div>
  Hello
</div>
`;
